src/arithmetic/arithmetic_expression.c

```diff
@@ -662,6 +699,9 @@ AR_ExpNode *AR_EXP_Clone(AR_ExpNode *exp) {
 }
 
 static inline void _AR_EXP_FreeOpInternals(AR_ExpNode *op_node) {
+	if(op_node->op.type == AR_OP_FUNC && op_node->op.f->bfree) {
+		op_node->op.f->bfree(op_node);
+	}
```

Comment:
If that's currently being used to free an `SI_VALUE` of type `PTR` representing an arithmetic expression?
if so, how did Dvir dealt with that in his `PATH` function(s) ?

Reply:
`T_PATH` is an `SIType`, so it has a free routine to be invoked in `SIValue_Free`.